### PR TITLE
Added a doc file(#3897)

### PR DESCRIPTION
**Pull request by singhvisha**
Added a reference.js file that contain explanation of few js keyword and their example.

(#3897)

### CODE REVIEW

**Comment by outofambit** [comment]
this is looking good! there's a couple formatting things we need to fix for the tests + linter to pass and i can help with that (probably in the next couple days)!

**Comment by limzykenneth** [comment]
The `package-lock.json` file is changed in this PR which is not expected. @singhvisha Did the npm version you had made those changes upon running `npm install`?

**Comment by stalgiag** [comment]
If it is helpful I find that using `npm ci` ([info link here](https://docs.npmjs.com/cli/ci.html)) instead can help when `npm install` is unexpectedly modifying the `package-lock.json` not sure if that is the precise difficulty here but just a thought.

Other than the problem with the package-lock this is looking pretty exciting. We will probably need to do a collective editing pass to make sure that the language is as accessible to beginners as possible. This is a really great beginning.

**Comment by limzykenneth** [comment]
Upon further inspection, the current `package-lock.json` is out of sync with `package.json` and so when running `npm install` it will repopulate `package-lock.json` with new data. I think the version committed here is correct but it's hard to trace down why the previous changes were made (most came from different commits). We may need to have a round of dependency updates along with `npm audit` very soon.

I'm ok to merge this PR as is.

File: src/core/reference.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
+/**
+ * file contains few JS library keywords and their examples.
+ */
+/*
+ * @Decleration  Let
```
Comment (stalgiag) [inline src/core/reference.js]:
```suggestion
 * @Declaration  Let
```
Small typo throughout. See [this page for spelling](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements).

File: src/core/reference.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+ * Declares a read-only named constant.
+ * Constants are block-scoped, much like variables defined using the 'let' statement.
+ * The value of a constant can't be changed through reassignment, and it can't be redeclared.
+ *
+ * Syntax:
+ * const name1 = value1, name2 = value2, ... nameN = valueN;
+ * @params name1,name2 ...,nameN   //The constant's name, which can be any legal identifier.
+ * value1,value2 ...,valueN    //The constant's value; this can be any legal expression, including a function expression.
+ *
+ * @exmaple
```
Comment (stalgiag) [inline src/core/reference.js]:
```suggestion
 * @example
```

**Comment by stalgiag** [comment]
Hi okay as promised I played around a bit and managed to figure out how to get this building the docs as desired. I am copying a small example here:
```
/**
 * @module Foundation
 * @submodule Declaration
 * @for p5
 */

/**
 *
 * Declares a block scope local variable, optionally initializing it to a value.
 *
 * @property  let
 *
 * @example
 * <div class="norender">
 * <code>
 * let myVariable = 1;
 * print(myVariable); // writes 1 to the console
 * myVariable = 2;
 * print(myVariable) // writes 2 to the console
 *
 * </code>
 * </div>
 *
 */

/**
 * @submodule Control Statement
 */

/**
 *
 * The if statement executes a statement if a specified condition is truthy.
 * If the condition is falsy, another statement can be executed
 *
 * @property  if-else
 *
 * @example
 * <div class="norender">
 * <code>
 * if (a > 0) {
 *  print('positive');
 * } else {
 *  print('negative');
 * }
 * </code>
 * </div>
 *
 */
```
If you put this into `reference.js` and run `npm run docs:dev` you should see the docs at your `localhost:9001`. These are separated by module and submodule. The naming of the module and submodule is up for discussion but I just wanted to get you started with something that builds. In general, I would say try to keep both language and examples as simple as possible which is something that @outofambit can probably help us figure out. Also, I would say that generally this PR doesn't need to be in its absolute perfect final form before merging since others such as @shubymao want to help with this work but it is important to figure out how to make sure that it builds correctly. Would you mind putting your work into a format similar to the one above and testing that you can see everything when running `npm run docs:dev`? Thank you for all of the hard work!

**Comment by singhvisha** [comment]
hi @stalgiag,what you want to reformat the code as you prescribed above and create an another Pull Request.

**Comment by stalgiag** [comment]
Hi @singhvisha whichever is easier for you! You can close this pull request and open another **or** you can modify the `core/reference.js` file and push to this branch.

**Pull request closed without merging**